### PR TITLE
News Api (Endpoints and Schema)

### PR DESCRIPTION
**Pull request by salujajaskeerat**
I have included News  model , router file and litle bit of controllers .
Although work is not much done.....

### CODE REVIEW

File: src/controllers/news.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    const user = await User.findById(req.payload.id);
+    if (user === null) {
+      throw createError(401, 'Unauthorized', 'Invalid Login Credentials');
+    }
+    const news = new News({
+      ...req.body,
+      createdBy: user._id,
+    });
+    await news.save();
+    res.send({news});
```
Comment (codeboy5) [inline src/controllers/news.ts:31]: create a JSON like it has been implemented in the event controller

**Comment by salujajaskeerat** [comment]
Added news api enpoints : Crud operation , report news etc

File: src/controllers/news.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
+/* eslint-disable @typescript-eslint/type-annotation-spacing */
+/* eslint-disable eol-last */
+/* eslint-disable prettier/prettier */
```
Comment (codeboy5) [inline src/controllers/news.ts:3]: Can you remove this statements, you don't need to disable anything, for prettier just running fixes everything automatically `npm run prettier-fix`. Let's follow these style guides strictly

File: src/controllers/news.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      res.send(trendNews)
+    }
+    else{
+      throw new Error('Query Not Matched')
+    }
+
+
+
+  } catch (e) {
+    res.status(500).send(e);
```
Comment (codeboy5) [inline src/controllers/news.ts:33]: You don't need to send the response here, just use `return next(e)` to the error to error handling middleware. You can check `user.ts` for reference. Same goes for all the functions.

File: src/controllers/news.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    const user = await User.findById(req.payload.id);
+    if (user === null || user.adminOf.length === 0) {
+      throw createError(401, 'Unauthorized', 'Invalid Login Credentials');
+    }
+    const news = new News({
+      ...req.body,
+      createdBy: user._id,
+    });
+    await news.save();
+    res.send({news});
```
Comment (codeboy5) [inline src/controllers/news.ts:70]: You can use the createResponse function here , it helps to keep the responses consistent.
`return res.send(createResponse('News Added Successfully', news));`

**Comment by codeboy5** [comment]
> Crud

I have added some comments, fix them. Everything else seems good to me. Good Work :)

**Comment by salujajaskeerat** [comment]
code modified as per requirements......
prettier-fix
used createResponse()
used next(error)

**Pull request merged**